### ISSUE TITLE
Include .editorconfig file in root folder

### ISSUE THREAD

**Issue by wesm87** [Varying-Vagrant-Vagrants/VVV#824]
This would help enforce some consistency, especially with new contributors. I was thinking something like the following:

```
root = true

[*]
indent_style = space
indent_size = 2
end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace = true
```

**Comment by cfoellmann** [comment]
I like the idea of .editorconfig
I am still hoping for support in Netbeans but a lot of other editors support it.
:+1:

**Comment by wesm87** [comment]
@cfoellmann That's a good point. The `Vagrantfile` and most of the `.sh` files are 2 spaces, but some of the `.sh` use tabs. So that was what originally inspired the idea. Looking at the Markdown files, though, it looks like they use 4 spaces, so maybe we'd want to add:

```
[*.md]
indent_size = 4
insert_final_newline = false
```

(The `insert_final_newline` setting is something I often see set to `false` for Markdown files)

**Comment by wesm87** [comment]
@cfoellmann Also have you tried this Netbeans plugin? https://github.com/welovecoding/editorconfig-netbeans

**Comment by jeremyfelt** [comment]
I like the idea of including an `.editorconfig`. We would need to make sure it only applies to VVV project files and that it accounts for markdown, shell, PHP, and other package configuration files.

**Comment by wesm87** [comment]
I wouldn't mind putting a PR together if we could define what standards we want to use for which file types. To exclude a directory, we can add a `.editorconfig` file with just the line `root = true` to disable our custom rules within that directory. We could also explicitly list every folder we want to include in each rule declaration, but that would be a lot more unwieldy and difficult to maintain.

With the exception of markdown and diff files, I'd say the following settings should be consistent across all file types:

```
end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace = true
```

So basically it's just a matter of what we want to use for the `indent_size` and `indent_style` for each file type. So far here's what I've come up with:

```
root = true


[*]
charset = utf-8
end_of_line = lf
trim_trailing_whitespace = true
insert_final_newline = true
indent_style = space
indent_size = 2

[*.{diff,md}]
insert_final_newline = false
trim_trailing_whitespace = false

[*.{conf,conf-sample}]
indent_size = 4

# Use WordPress-style tabs for PHP files
[*.php]
indent_style = tab
indent_size = 4
```

If anyone has any suggestions feel free to chime in.
